pulsar-client-cpp/docker-tests.sh

```diff
@@ -58,4 +60,26 @@ done
 
 # Start 2 Pulsar standalone instances (one with TLS and one without)
 # and execute the tests
-$DOCKER_CMD bash -c "cd /pulsar/pulsar-client-cpp && ./run-unit-tests.sh ${tests}"
+set +e
+DISABLE_COLOR_OUTPUT=""
+if [ "$GTEST_COLOR" = "no" ]; then
+  DISABLE_COLOR_OUTPUT="| cat"
+fi
+$DOCKER_CMD bash -c "cd /pulsar/pulsar-client-cpp && ./run-unit-tests.sh ${tests} $DISABLE_COLOR_OUTPUT"
+RES=$?
+if [ $RES -ne 0 ]; then
```

Comment:
This check would never success. The error code returned by `$?` is the docker container's exit code, not the command's exit code. Even if `run-unit-tests.sh` exited with a non-zero code, the `RES` would still and always be 0.

Reply:
Based on documentation, docker run will return the exit code. Docs: https://docs.docker.com/engine/reference/run/#exit-status

Reply:
Yeah, it's a little weird that when I run the `docker run` with `run-unit-tests.sh` that failed, then print `echo $?`, it printed `1`. However, when I run `docker-tests.sh` and added a `echo $RES` after line 69, the tests failed but it printed `0`. I'll continue to dig into this issue.